src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Cache.cs

```diff
@@ -114,17 +139,17 @@ public override bool Equals(object obj)
 
             public bool Equals(CachedCodeEntryKey other)
             {
-                return this == other;
+                return _pattern.Equals(other._pattern) && _options == other._options && _cultureKey.Equals(other._cultureKey);
```

Comment:
Do we have a test verifying that we check options and culture in Equals/GetHashcode? I realize you didn't change this.

Reply:
No. Eventually we need access to the cache internals. I'm trying to make `[assembly:InternalsVisibleTo]` as done in `System.IO.Pipelines`. Some publickey there though. And adding more detailed tests.

Reply:
If it's just one or two methods /types then reflection is okay.

Reply:
Fixed